rusoto/core/src/pagination.rs

```diff
@@ -0,0 +1,76 @@
+//! Traits and helperfunctions for dealing with paged responses
+
+use futures::{stream, Stream, TryStreamExt, Future};
+use crate::{RusotoError};
+use std::pin::Pin;
+
+/// A stream of results returned over multiple pages by amazon
+pub type RusotoStream<I, E> = Pin<Box<dyn Stream<Item = Result<I, RusotoError<E>>>>>;
+
+/// Trait representing a request that can have a token added to fetch subsequent pages
+pub trait PagedRequest {
+    /// The type of the paging token, usuaully `Option<String>` but sometimes a tuple or something mor exotic
+    type Token;
+
+    /// Set the token to get the next page
+    /// See [PagedOutput::pagination_token]
+    fn with_pagination_token(self, key: Self::Token) -> Self;
```

Comment:
Could be `set_pagination_token(&mut self, key: Self::Token);

Reply:
Fixed!, I was using the builder style naming becasue it made sense in the generated code at some point but this is better.